### PR TITLE
refactor: Use FormattingConfig in format.py

### PR DESCRIPTION
**Pull request by yoomlam**
## Ticket

Related to #128

## Changes

Refactor to use `FormattingConfig` and `BemFormattingConfig` instead of `if-then` blocks.

### CODE REVIEW

File: app/tests/src/test_format.py [hunk 1]
```diff
@@ -264,35 +267,32 @@ def test__return_citation_link():
...
     )
 
     assert (
-        format_web_subsections(0, 0, [], [], "List intro sentence: \n- item 1\n- item 2")
+        build_accordions([], "List intro sentence: \n- item 1\n- item 2", config)
         == "<div><p>List intro sentence: </p>\n<ul>\n<li>item 1</li>\n<li>item 2</li>\n</ul></div>"
     )
 
-    chunks_with_scores[0].chunk.document.name = "Your State Disability Insurance (SDI)"
-    chunks_with_scores[1].chunk.document.name = "Your State Disability Insurance (SDI): Another"
```
Comment (yoomlam) [inline app/tests/src/test_format.py:292]: This refactoring revealed that `chunks_with_scores` is not relevant to `format_web_subsections` (now `build_accordions`)

File: app/tests/src/test_format.py [hunk 2]
```diff
@@ -264,35 +267,32 @@ def test__return_citation_link():
...
     chunk_list[1].page_number = 3
 
-    bem_link = return_citation_link(chunk_list[0], "BEM")
+    bem_link = BemFormattingConfig().return_citation_link(chunk_list[0])
 
     assert "Open document to page 3" in bem_link
     assert "Source" not in bem_link
 
-    web_link = return_citation_link(chunk_list[1], "EDD")
+    web_link = FormattingConfig().return_citation_link(chunk_list[1])
```
Comment (yoomlam) [inline app/tests/src/test_format.py:275]: It is now clear that we only use 2 FormattingConfigs

File: app/src/chat_engine.py
Comment (yoomlam) [inline app/src/chat_engine.py:1]: Note on line 39, we're using the same LLM system prompt for all chat engines! This means all engines are expecting to talk about California and EDD.

Comment (yoomlam) [reply app/src/chat_engine.py:1]: #130

**Review by KevinJBoyer — APPROVED** [review]
LGTM, thank you!

**Pull request merged**